### PR TITLE
Add macOS sysroot and LFLAGS handling for Conda builds

### PR DESCRIPTION
**Pull request by yuxuanzhuang**
I used `build-locally.py` to build mac arm version and it passed locally.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -2,7 +2,17 @@
 cd ${SRC_DIR}/src
 source ../source.apache
 
+if [[ "$target_platform" == osx-* ]]; then
```
Comment (orbeckst) [inline recipe/build.sh:5]: This pattern matching works?!

**Comment by orbeckst** [comment]
@conda-forge-admin, please rerender

**Review by orbeckst — CHANGES_REQUESTED** [review]
The linter wants a static dependency. Can you look into adding it or is it not necessary??

Not all tests passed—not sure why. I’d expect all to be green before merging.

File: recipe/build.sh (hunk 1 above)
Comment (yuxuanzhuang) [reply recipe/build.sh:5]: The `target_platform` is set in yaml files e.g. https://github.com/conda-forge/hole2-feedstock/blob/45b11db2967499dff5afe5b3ad21b4d0fecfdabb/.ci_support/osx_64_.yaml#L17 if that's what you mean?

**Comment by yuxuanzhuang** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh (hunk 1 above)
Comment (orbeckst) [reply recipe/build.sh:5]: I meant the `*` – but I may have misunderstood what it does. In any case, wasn't really relevant.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,6 +19,7 @@ requirements:
   build:
     - {{ compiler('fortran') }}
     - {{ compiler('c') }}
+    - {{ stdlib('c')}}
```
Comment (orbeckst) [inline recipe/meta.yaml]:
```suggestion
    - {{ stdlib('c') }}
```

**Review by orbeckst — APPROVED** [review]
CI is actually green so let's try it.

**Pull request merged**